fix: add conceal property in elements/text rendering

glwbr opened the pull request:
## Description

This PR fixes the conceal style in renderText, which wasn't working before. Although Conceal was defined in StylePrimitive, it wasn't being used during rendering.

Closes: [Using a custom style with conceal:true does not hide element #645](https://github.com/charmbracelet/glow/issues/645)

## Changes

- renderText now checks the Conceal property
- if `"conceal"` is true, the text is skipped during rendering

## Testing

Tested with a minimal custom style that uses the conceal property for the elements:

```json
{
  "document": {
    "block_prefix": "\n",
    "block_suffix": "\n",
    "color": "#d0d0d0",
    "margin": 0
  },
  "paragraph": {},
  "heading": {
    "color": "#87d7ff",
    "bold": true
  },
  "link": {
    "color": "#00afd7",
    "underline": true,
    "conceal": true
  },
  "link_text": {
    "color": "#00afd7",
    "underline": true
  },
  "code": {
    "color": "#d7d7ae",
    "conceal": true
  },
  "image": {
    "color": "#00afd7",
    "conceal": true
  },
  "image_text": {
    "color": "#00afd7",
    "underline": true
  }
}
```

```md
# Concealment Test

## Links

This is a [test link](https://example.com) that should conceal the URL.

## Code

Some `inline code` that we might want to conceal.

## Images

![Image alt text](https://example.com/image.jpg)
```

andreynering reviewed (approved):
It works for me, but I noticed some extra white space that we could try to address somehow.

It may be fine to merge as is and fix later, as this is already better than before.

Before:

<img width="628" alt="Screenshot 2025-04-09 at 10 28 20" src="https://github.com/user-attachments/assets/b3f9a4d4-aa3c-4e35-8866-4c952242e3b5" />

After:

<img width="464" alt="Screenshot 2025-04-09 at 10 28 35" src="https://github.com/user-attachments/assets/bd41f810-2536-46eb-9ce6-d89fb845b626" />

meowgorithm commented:
For some background with this one, I believe the original thought behind the conceal property was to offer support for the rarely-supported ANSI conceal property `\x1b[8m`, which Glamour may or may not be applying under the hood.

If we go ahead with this one we should probably fix the spacing and assure it works as expected across the board.

glwbr commented:
@meowgorithm When applying the escape sequence, the result will be the "hidden" text but with the element space maintained. Is this the expected behavior?

![image](https://github.com/user-attachments/assets/0f9bc180-c8e4-4c23-ad63-0d536219c4af)

For link/link_text and image/image_text elements, I addressed the issues directly in the element renderer:

```go
func (e *LinkElement) renderHrefPart(w io.Writer, ctx RenderContext) error {
	linkTextStyle := ctx.options.Styles.LinkText
	isTextConcealed := linkTextStyle.Conceal != nil && *linkTextStyle.Conceal

	linkStyle := ctx.options.Styles.Link
	isLinkConcealed := linkStyle.Conceal != nil && *linkStyle.Conceal

	prefix := ""
	if !e.SkipText && !isTextConcealed && !isLinkConcealed {
		prefix = " "
	}

	u, err := url.Parse(e.URL)
	if err == nil && "#"+u.Fragment != e.URL { // if the URL only consists of an anchor, ignore it
		if !isLinkConcealed {
			el := &BaseElement{
				Token:  resolveRelativeURL(e.BaseURL, e.URL),
				Prefix: prefix,
				Style:  linkStyle,
			}
			if err := el.Render(w, ctx); err != nil {
				return err
			}
		}
	}
	return nil
}
}
```
